relay-profiling/src/cocoa.rs

```diff
@@ -139,29 +132,35 @@ impl CocoaProfile {
     }
 }
 
-pub fn expand_cocoa_profile(payload: &[u8]) -> Result<Vec<Vec<u8>>, ProfileError> {
-    let profile = parse_cocoa_profile(payload)?;
-    let mut items: Vec<Vec<u8>> = Vec::new();
+fn parse_profile(payload: &[u8]) -> Result<CocoaProfile, ProfileError> {
+    let mut profile: CocoaProfile =
+        serde_json::from_slice(payload).map_err(ProfileError::InvalidJson)?;
 
-    if profile.transactions.is_empty() && profile.has_transaction_metadata() {
-        match serde_json::to_vec(&profile) {
-            Ok(payload) => items.push(payload),
-            Err(_) => {
-                return Err(ProfileError::CannotSerializePayload);
-            }
-        };
+    if profile.transactions.is_empty() && !profile.has_transaction_metadata() {
+        return Err(ProfileError::NoTransactionAssociated);
+    }
+
+    profile.remove_single_samples_per_thread();
 
-        return Ok(items);
+    if profile.sampled_profile.samples.is_empty() {
+        return Err(ProfileError::NotEnoughSamples);
+    }
+
+    if profile.transactions.is_empty() && profile.has_transaction_metadata() {
```

Comment:
This logic here is different than in the android case. In android case, this early return doesn't exist. Is that intentional?

Reply:
Yes, it's intentional because in the Android case, we still need to go through the `parse` logic to extract an `AndroidTrace` even if the transaction is set on top level fields.